picard/ui/itemviews/basetreeview.py

```diff
@@ -567,3 +568,82 @@ def moveCursor(self, action, modifiers):
     @property
     def default_drop_target(self):
         return None
+
+    def setup_find_box(self):
+        self.find_box = FindBox(self)
+        self.find_box.findChanged.connect(self.filter_items)
+
+        self.find_box.hide()  # Hide the find box initially
+
+        return self.find_box
+
+    def filter_items(self, text, filters):
+        if not text:  # When text is empty, show all items
+            self._restore_all_items()
+            return
+
+        text = text.lower()
+        self._filter_tree_items(self.invisibleRootItem(), text, filters)
```

Comment:
Instead of passing `text.lower()` to `_filter_tree_items()`, pass `text` and do the conversion inside  `_filter_tree_items()`.